src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -467,5 +471,218 @@ public override T GetResult<T>()
                 return result.ToObject<T>(this.jsonSerializer);
             }
         }
+
+        private class BufferTextWriter : TextWriter
```

Comment:
A future iteration of this PR removes this class in favor of consuming a [more complete and tested version from Nerdbank.Streams](https://github.com/AArnott/Nerdbank.Streams/pull/71/files#diff-dee8a6ac9111d50058f2955c13f847da).